src/rgw/rgw_log.cc

```diff
@@ -226,6 +225,18 @@ static void log_usage(struct req_state *s, const string& op_name)
   utime_t ts = ceph_clock_now();
 
   usage_logger->insert(ts, entry);
+
+  if(s->cct->_conf->rgw_enable_usage_log_at_subuser_level) {
+    const auto subuser = s->auth.identity->get_subuser_name();
+
+    if (subuser) {
+      std::string sname = u + ":" + *subuser;
+      ldout(s->cct, 5) << "log subuser entry: " << sname << dendl;
+      rgw_usage_log_entry sentry(sname, p, bucket_name);
```

Comment:
Answering the question about *multi-tenant* buckets from [your comment](https://github.com/ceph/ceph/pull/13929/files#r108873785): `bucket_name` here doesn't contain the tenant part. It directly comes from `req_state::bucket_name` which lacks the information about tenancy as it's stored separately as `req_state::bucket_tenant`. This translates into relying entirely on the tenant part of `rgw_user`. It's serialised by `to_str()` into `<tenant>:<user>` which is logically undistinguishable from `<user>:<subser>` in the case of the global, empty tenant. This ambiguity might lead to wrong accounting which a severe issue.